### PR TITLE
PIM-6812: Change the message when delete an attribute as variant axis

### PR DESCRIPTION
**Pull request by aRn0D**
**Description (for Contributor and Core Developer)**

We cannot delete an attribute use as family variant axis (business rule). When the front send a request to remove an attribute, we check if the attribute belongs to a family variant before. 

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | yes
| Added Behats                      | -
| Added integration tests           | yes
| Changelog updated                 | yes
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Query/AttributeIsAnFamilyVariantAxis.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+use Pim\Component\Catalog\Model\VariantAttributeSetInterface;
+
+/**
+ * Checks if an attribute is used as family variant axis
+ *
+ * @author    Arnaud Langlade <arnaud.langlade@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class AttributeIsAnFamilyVariantAxis
```
Comment (anaelChardan) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Query/AttributeIsAnFamilyVariantAxis.php]: `A` Family Variant Axis ;)

File: CHANGELOG-2.0.md [hunk 2]
```diff
@@ -4,6 +4,11 @@
 
 - PIM-7032: Fix close button when clicking on "Compare/Translate" option on the product edit form
 - PIM-7063: Add validation for AttributeGroup - cannot remove AttributeGroup containing attributes and cannot remove AttributGroup "other"
+- PIM-6812: Change the message when delete an attribute as variant axis
+
+## BC breaks
+
+- Changes the constructor of Pim\Bundle\EnrichBundle\Controller\Rest\AttributeController` add `Doctrine\ORM\EntityManagerInterface` and `Symfony\Component\Translation\TranslatorInterface`
```
Comment (anaelChardan) [inline CHANGELOG-2.0.md]: `Change`

File: CHANGELOG-2.0.md [hunk 3]
```diff
@@ -4,6 +4,11 @@
 
 - PIM-7032: Fix close button when clicking on "Compare/Translate" option on the product edit form
 - PIM-7063: Add validation for AttributeGroup - cannot remove AttributeGroup containing attributes and cannot remove AttributGroup "other"
+- PIM-6812: Change the message when delete an attribute as variant axis
+
+## BC breaks
+
+- Change the constructor of Pim\Bundle\EnrichBundle\Controller\Rest\AttributeController` add `Doctrine\ORM\EntityManagerInterface` and `Symfony\Component\Translation\TranslatorInterface`
```
Comment (SamirBoulil) [inline CHANGELOG-2.0.md]: missing '`'

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php [hunk 4]
```diff
@@ -98,7 +108,9 @@ public function __construct(
         AttributeFactory $factory,
         UserContext $userContext,
         LocalizerInterface $numberLocalizer,
-        NormalizerInterface $lightAttributeNormalizer
+        NormalizerInterface $lightAttributeNormalizer,
```
Comment (SamirBoulil) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php]: maybe BCing a 2015 class is not wise :/

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php [hunk 5]
```diff
@@ -281,10 +295,23 @@ public function postAction(Request $request, $identifier)
      *
      * @AclAncestor("pim_enrich_attribute_remove")
      */
-    public function removeAction($code)
+    public function removeAction(string $code): JsonResponse
```
Comment (SamirBoulil) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php]: :+1: add declare strict types ?

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php (hunk 4 above)
Comment (aRn0D) [reply src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php]: We don't have choice

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php (hunk 5 above)
Comment (docteurklein) [reply src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php]: `declare(strict_types=1)` only has impacts on the caller.
Here it's a controller, so the caller is the `HttpKernel::handle()` method, which we have no control on.

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php [hunk 6]
```diff
@@ -113,6 +127,8 @@ public function __construct(
         $this->userContext = $userContext;
         $this->numberLocalizer = $numberLocalizer;
         $this->lightAttributeNormalizer = $lightAttributeNormalizer;
+        $this->translator = $translator;
+        $this->attributeIsAnFamilyVariantAxisQuery = $attributeIsAnFamilyVariantAxisQuery;
```
Comment (anaelChardan) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php]: attributeIsAFamilyVariantAxisQuery, sorry, I forgot that on my previous review.

**Comment by jjanvier** [comment]
CI was green
I rebased only for a conflict with the CHANGELOG

**Pull request merged**